lib/diff.ml

```diff
@@ -113,8 +120,50 @@ let type_item ~typing_env ~name ~reference ~current =
       in
       match (type_coercion1 (), type_coercion2 ()) with
       | None, None -> None
-      | _, _ ->
-          Some (Type { tname = name; tdiff = Modified { reference; current } }))
+      | _, _ -> (
+          match (reference.type_kind, current.type_kind) with
+          | Type_record (ref_label_lst, _), Type_record (cur_label_lst, _) ->
+              Some
+                (Type
+                   {
+                     tname = name;
+                     tdiff =
+                       Modified
+                         (Record
+                            (modified_record_type ~ref_label_lst ~cur_label_lst));
+                   })
+          | _, _ ->
+              Some
+                (Type
+                   {
+                     tname = name;
+                     tdiff = Modified (Any { reference; current });
+                   })))
+
+and modified_record_type ~ref_label_lst ~cur_label_lst =
+  let ref_lbls = extract_lbls ref_label_lst in
+  let curr_lbls = extract_lbls cur_label_lst in
+  String_map.merge
+    (fun name ref cur ->
+      match (ref, cur) with
+      | None, None -> None
+      | Some ref, None -> Some { lname = name; ldiff = Removed ref }
+      | None, Some cur -> Some { lname = name; ldiff = Added cur }
+      | Some ref, Some cur ->
+          if ref.ld_type = cur.ld_type then None
```

Comment:
I think we should be more careful here: The type may change without introducing a breaking change.

For instance:
```
type r = { a : int }
```
becoming
```
type a = int
type r = { a : a }
```

Reply:
Yes definitely, this needs to be checked in a similar way to how we diff values!

Polymoprhic equality probably won't even work in the simple cases here anyway as the `core_type` representation is stateful and holds a lot of information beyond the type representation.

Reply:
If I am not mistaken, there should be a way to lookup a `type_expr` in the typing_env, and check if the alias resolves to the same type, right ?